test/engine.spec.js

```diff
@@ -1,552 +1,307 @@
-
-const removeVersions = (engine) => {
-    if(engine.versions) {
-        delete engine.versions;
+
+
+import mockFS from 'mock-fs';
+import sinon from 'sinon';
+import chai, { expect } from 'chai';
+import sinonChai from 'sinon-chai-es';
+chai.use(sinonChai);
+import semver from 'semver';
+
+import { Engine } from '../src/common/engine.js';
+
+// eslint-disable-next-line no-unused-vars
+import Version from '../src/common/version.js';
+
+const myVersions = {
+    'c:\\test': {
+        afile: 'hello'
+        , nvm: {
+            'v8.11.1':{
+                'node64.exe': ''
+                , node56: {}
+            }
+            , 'v0.0.1':{
+                'node64.exe': ''
+                , 'node.exe': ''
+                , node56: {}
+            }
+            , 'v2.10.22':{
+                'node.exe': ''
+                , node56: {}
+            }
+            , 'v10.23.0':{
+                'node64.exe': ''
+                , node56: {}
+                , node: {}
+            }
+            , 'v99.99.99':''
+        }
+        , '.nvm': {
+            versions: {
+                node: {
+                    'v10.03.0':{
+                        bin: {
+                            node64: ''
+                            , node: ''
+                            , node56: {}
+                        }
+                    }
+                    , 'v0.0.1':{
+                        bin: {
+                            node64: ''
+                            , node: mockFS.file({
+                                content: ''
+                                , mode: 0o666
+                            })
+                            , node56: {}
+                        }
+                    }
+                    , 'v2.10.11':{
+                        bin: {
+                            node64: ''
+                            , node:  mockFS.file({
+                                content: ''
+                                , mode: 0o755
+                            })
+                            , node56: {}
+                        }
+                    }
+                    , 'v12.0.0':{
+                        bin: {
+                            node: ''
+                            , node64: ''
+                            , node56: {}
+                        }
+                    }
+                    , 'v10.14.0':{
+                        bin: {
+                            node64: ''
+                            , node56: {}
+                        }
+                    }
+                    , 'v99.99.99':''
+                }
+            }
+        }
     }
 };
 
-const engine = require('../src/common/engine');
-
-describe('Engine Module', function() {
-    let logSpy;
+describe('Engine Class', function() {
+    const mockFSOptions = {
+        env: { NVM_HOME: 'c:\\test\\nvm' }
+    };
     beforeEach(function() {
-        logSpy = sinon.stub();
+        mockFS(myVersions);
+    });
```

Comment:
mocha specifically discourages the use of arrow functions because its not possible to obtain the mocha context, see https://mochajs.org/#arrow-functions